ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/transaction/TransactionSignatureService.java

```diff
@@ -57,12 +58,22 @@ List<List<DigitalSignatureAndMetadata>> signBatch(
      * The underlying verification service signals the verification failures with different exceptions.
      * {@link DigitalSignatureVerificationService}
      *
-     * @param transaction The original transaction.
+     * @param transaction           The original transaction.
      * @param signatureWithMetadata The signature to be verified.
+     * @param publicKey             The public key to verify against. It should match with signatureWithMetadata's keyId.
      * @throws RuntimeException if the signature could not be verified.
      */
     void verifySignature(
             @NotNull final TransactionWithMetadata transaction,
-            @NotNull final DigitalSignatureAndMetadata signatureWithMetadata
+            @NotNull final DigitalSignatureAndMetadata signatureWithMetadata,
+            @NotNull final PublicKey publicKey
+    );
+
+    // TODO this needs internally in a few places, but maybe it could be kept internally? TransactionSignatureServiceInternal maybe?
```

Comment:
So you have exposed this, because it needs to be also used outside `TransactionSignatureServiceImpl`. However, I think it feels like an internal utility.

Reply:
Yes. In the meantime I moved it into a `TransactionSignatureServiceInternal` interface in runtime-os.
The impl part stayed the same and that same impl, implements this new interface too.
I am not 100% sure how this plays with OSGi...
If it does not we can still keep it on the public API, on another internal service or somewhere else.